src/sources/syslog.rs

```diff
@@ -349,7 +388,15 @@ fn enrich_syslog_event(event: &mut Event, host_key: &str, default_host: Option<B
         .get("timestamp")
         .and_then(|timestamp| timestamp.as_timestamp().cloned())
         .unwrap_or_else(Utc::now);
-    log.insert(log_schema().timestamp_key(), timestamp);
+    match log_namespace {
+        LogNamespace::Vector => {
+            log.insert(metadata_path!("vector", "ingest_timestamp"), Utc::now());
```

Comment:
I believe this can be combined with the `source_type` above to just use `insert_standard_vector_source_metadata()`

Reply:
On reflection, I think the timestamp should be handled differently. I've changed it so that it inserts `ingest_timestamp` as all the others. 

Then for the timestamp that is taken from the syslog message, for the Legacy namespace it inserts it or it inserts `Utc::now`. For the Vector namespace it inserts it only if it exists in the message.